src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SNITcpHandle.cs

```diff
@@ -532,42 +526,21 @@ public override void SetAsyncCallbacks(SNIAsyncCallback receiveCallback, SNIAsyn
         /// <returns>SNI error code</returns>
         public override uint SendAsync(SNIPacket packet, SNIAsyncCallback callback = null)
         {
-            SNIPacket newPacket = packet;
-
-            _writeTaskFactory.StartNew(() =>
```

Comment:
After this change the fields 
```C#
private readonly TaskScheduler _writeScheduler;
private readonly TaskFactory _writeTaskFactory;

``` 
are not needed

Reply:
fixed.